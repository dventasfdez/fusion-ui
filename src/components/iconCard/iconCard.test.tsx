import React from 'react';
import renderer from 'react-test-renderer';
import IconCard, {IconCardIcon, IconCardHeader, IconCardBody, IconCardFooter} from './iconCard';

const iconCardExample = (props?: any) => (
  <div>
    <IconCard {...props}>
      <IconCardIcon>
        <span className="material-icons">collections</span>
      </IconCardIcon>
      <IconCardHeader>
        <h4>This is a title</h4>
      </IconCardHeader>
      <IconCardBody>Lorem ipsum dolor, sit amet consectetur adipisicing elit. Sed, excepturi.</IconCardBody>
      <IconCardFooter>
        <a href="">link</a>
      </IconCardFooter>
    </IconCard>
  </div>
);

/**
 * ICON CARD
 */
describe('Icon card type tests', () => {
  test('Icon card component should render', () => {
    const component = renderer.create(iconCardExample({className: 'tag-ds'}));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Icon card selected component should render', () => {
    const component = renderer.create(iconCardExample({selected: true}));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
