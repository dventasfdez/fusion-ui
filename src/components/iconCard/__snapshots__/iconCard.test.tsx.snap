// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon card snapshots Icon card component should render 1`] = `
<div>
  <div
    class="card_icon "
    id=""
  >
    <div
      class="card-icon "
      id=""
    >
      <span
        class="material-icons"
      >
        collections
      </span>
    </div>
    <div
      class="card_icon-content"
    >
      <div
        class="card-header "
      >
        <h4>
          This is a title
        </h4>
      </div>
      <div
        class="card-body "
      >
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Sed, excepturi.
      </div>
      <div
        class="card-footer "
      >
        <a
          href=""
        >
          link
        </a>
      </div>
    </div>
  </div>
</div>
`;
